apps/nrm/behaviours/delete-on-change.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+/**
+ * Delete supplied fields from session
+ *
+ * @param {object} req - request object
+ * @param {object} config - object containing configuration details for function
+ * for example config = { currentField: 'pv-name',  deleteFields: ['pv-dob'] }
+ *
+ * @returns {void}
+ */
+const deleteFieldsFromSession = (req, config) => {
+  const visited = req.sessionModel.get('steps');
+  const { deleteFields, currentField } = config;
+
+  if (visited && visited.indexOf(req.form.options.route) > -1) {
+    if (req.form.values[config.currentField] !== req.sessionModel.get(currentField)) {
+      // delete each field from the session
+      deleteFields.forEach(field => {
+        req.sessionModel.unset(field);
+      });
+
+      // Required for the back button
+      req.sessionModel.set('steps', visited);
```

Comment:
Are you sure you need this? I would have thought, It should be already in the req

Reply:
Nope, I'll remove that.